### PR TITLE
Cross build images and standard push/tag targets

### PR DESCRIPTION
**Pull request by deitch**
## Description
Support cross-build of images, and standard `push`, `tag-images`, `image`. `build` targets

cc @fasaxc

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,42 +1,74 @@
...
 
-.PHONEY: clean
+.PHONY: clean image
 
 # These variables can be overridden by setting an environment variable.
 LOCAL_IP_ENV?=$(shell ip route get 8.8.8.8 | head -1 | cut -d' ' -f8)
 
 CONFD_REPO?=calico/confd
 CONFD_VER?=v3.1.1
+# problem is that calico/conf does not yet have for other archs, and archless is amd64
```
Comment (fasaxc) [inline Makefile]: Typo: calico/conf

File: Makefile [hunk 2]
```diff
@@ -1,42 +1,74 @@
...
+default: clean image
 
-default: clean calicorr.created
+# standard build target, but nothing to do
+build:
 
-calicorr.created: $(BUILD_FILES) dist/confd
-	docker build $(DOCKER_EXTRA_BUILD_ARGS) -t calico/routereflector$(ARCHTAG) -f Dockerfile$(ARCHTAG) .
+image: calicorr.created
+calicorr.created: $(BUILD_FILES) dist/confd-$(ARCH)
```
Comment (fasaxc) [inline Makefile]: Probably need the arch in the name of the .created file or the first arch you build will prevent others from being built.

File: Makefile [hunk 3]
```diff
@@ -1,42 +1,74 @@
...
+	docker build $(DOCKER_EXTRA_BUILD_ARGS) -t $(CONTAINER_NAME):latest-$(ARCH) -f Dockerfile.$(ARCH) .
```
Comment (fasaxc) [inline Makefile]: Please can you add the usual "if it's amd64, retag as `:latest`" bit?

**Review by fasaxc — CHANGES_REQUESTED** [review]
Thanks @deitch, just a couple of minor points from me.

File: Makefile (hunk 1 above)
Comment (deitch) [reply Makefile]: Will fix.

File: Makefile (hunk 2 above)
Comment (deitch) [reply Makefile]: Will add. Good point.

File: Makefile (hunk 3 above)
Comment (deitch) [reply Makefile]: Sure.

File: Makefile (hunk 2 above)
Comment (deitch) [reply Makefile]: I think I will make a variable out of it.

**Comment by deitch** [comment]
All three comments addressed. Please review to make sure I got them correctly after CI passed, @fasaxc

**Comment by deitch** [comment]
Ugh, looks like I need a default archless `calicor.created` target for semaphore. Will add. We can address that later when we do the `make ci`/`make cd` work in the next phase.

**Comment by deitch** [comment]
Added `calicorr.created` to point to `calicorr.created-$(ARCH)`, which is really just the moral equivalent of `make image`. But the standard `image` target didn't exist before this PR, so Semaphore didn't use it.

Let's get this one through, then when we do the `make ci` standard we can clean those targets up.

**Comment by deitch** [comment]
CI passes now. Glad it caught it or something else would gave broken. :-)

**Review by fasaxc — APPROVED** [review]
LGTM

**Pull request merged**